languages/python/sqlalchemy-oso/sqlalchemy_oso/session.py

```diff
@@ -0,0 +1,155 @@
+"""SQLAlchemy session classes and factories for oso."""
+from sqlalchemy import event
+from sqlalchemy.orm.query import Query
+from sqlalchemy.orm import sessionmaker, Session
+from sqlalchemy import orm
+
+from oso import Oso
+
+from sqlalchemy_oso.auth import authorize_model_filter
+
+
+@event.listens_for(Query, "before_compile", retval=True)
+def _before_compile(query):
+    """Enable before compile hook."""
+    return _authorize_query(query)
```

Comment:
TODO: test that returning `None` here (e.g., the check on line 24, below) does nothing to the provided query.